packages/core/src/dataset-cache.ts

```diff
@@ -225,18 +226,20 @@ export class AsyncDataCache<SemanticKey extends RecordKey, CacheKey extends Reco
                 return resolvedCacheData;
             }
         } else {
-            promise = getter();
+            const abort = new AbortController();
```

Comment:
I don't think AbortControllers are all that expensive, but this will be called often.

Should be fine, but if they are expensive, that could be an issue!

Reply:
they have not seemed to be expensive in my testing so far - this change does use a lot more of them though - the old system could only cancel on a per-item basis (one Item -> many cache entries), but for the fine grained sharing and prioritization, using an abort signal per cache entry made more sense

Reply:
also also - even though we supported aborting fetches in the previous cache, it often wasnt connected, due to its poor behavior. You'd cancel a frame, that would abort a fetch for item X, and then the very next frame would fire up a fetch for item X :(  ... that should not happen anymore!